File: python/7 kyu/274.py
# https://www.codewars.com/kata/56f3a1e899b386da78000732/train/python

# Passed

def partlist(arr):
    parts_arr = []
    for idx in range(1, len(arr)):
        first_part = " ".join(arr[:idx])
        second_part = " ".join(arr[idx:])
        part = (first_part, second_part)
        parts_arr.append(part)
        
    return parts_arr

output = partlist(["cdIw", "tzIy", "xDu", "rThG"])
print(output)

File: nodejs/8 kyu/274.js
// https://www.codewars.com/kata/55eea63119278d571d00006a/train/javascript

// Passed

function nextId(ids) {
  let idsSet = new Set(ids);
  let smallestUnusedId = 0;
  while (idsSet.has(smallestUnusedId)) {
    smallestUnusedId++;
  }

  return smallestUnusedId;
}

const output = nextId([1, 2, 0, 2, 3, 5]);
console.log(output);